utils/update_arch_timings.py

```diff
@@ -32,37 +32,74 @@ def remove_site_number(site):
     return site
 
 
-def get_cell_types_and_instance(bel, location, site, bels):
+def gen_all_possibilities(pattern):
+    """
+    Generates all possible combinations of a pattern if it contains a
+    wildcard string in braces eg. "LUT[ABCD]" will yield in "LUTA", "LUTB"
+    and so on.
+
+    >>> list(gen_all_possibilities("LUT"))
+    ['LUT']
+
+    >>> list(gen_all_possibilities("LUT[ABCD]"))
+    ['LUTA', 'LUTB', 'LUTC', 'LUTD']
+    """
+
+    # Match the regex
+    match = re.match(r"(.*)\[([A-Za-z0-9]+)\](.*)", pattern)
+
+    # Generate combinations
+    if match is not None:
+        for c in match.group(2):
+            yield match.group(1) + c + match.group(3)
+
+    # Not a regex
+    else:
+        yield pattern
```

Comment:
shouldn't this be `return`? This is not a partial result

Reply:
No, that is correct. If the pattern does not contain a wildcard then it is considered to be a single name of a CELLTYPE.INSTANCE and is returned directly.

Reply:
this is exactly what I'm saying. If we return the value directly it should be done with `return`. `yield` is confusing here.

Reply:
But this function is used as a generator, it cannot return anything. It must yield even if it is a single item.

Reply:
ok